aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -135,6 +162,29 @@ TORCH_META_FUNC(fmin) (const Tensor& self, const Tensor& other) {
     build_binary_op(maybe_get_output(), self, other);
 }
 
+void comparison_op_check(const Tensor& self, const Tensor& other) {
+  // Validate that is possible to convert zero-dim tensor's dtype to other dtype
+  // without overflow
+  if (self.scalar_type() != other.scalar_type()) {
+    if (self.dim() != 0 && other.dim() == 0) {
+      native::check_convert(other.item(), self.scalar_type());
+    } else if (self.dim() == 0 && other.dim() != 0) {
+      native::check_convert(self.item(), other.scalar_type());
+    }
```

Comment:
Note to reviewers: this is ported code that cannot be deleted (and is not otherwise modified); hopefully to be deleted next PR

Reply:
@ngimel this logic looks potentially inefficient: if I have a CUDA scalar and I pass it to a comparison operation, I'm gonna sync for not very good reason.

Reply:
why will be be able to remove it in a future PR?

Reply:
no, as currently specced this is impossible to remove unless you turn it into an async cuda assert

Reply:
FYI, now people are rightfully complaining about it (and sorry I missed the original ping a year ago, lol) D36012582